packages/tldraw/src/lib/shapes/text/RichTextArea.tsx

```diff
@@ -0,0 +1,242 @@
+import { EditorView } from '@tiptap/pm/view'
+import { EditorEvents, EditorProvider, JSONContent } from '@tiptap/react'
+import {
+	Editor,
+	TLRichText,
+	TLShapeId,
+	preventDefault,
+	stopEventPropagation,
+	useEditor,
+	useUniqueSafeId,
+} from '@tldraw/editor'
+import React, { useEffect, useState } from 'react'
+
+/** @public */
+export interface TextAreaProps {
+	isEditing: boolean
+	text?: string
+	shapeId: TLShapeId
+	richText?: TLRichText
+	handleFocus(): void
+	handleBlur(): void
+	handleKeyDown(e: KeyboardEvent): void
+	handleChange(changeInfo: { plaintext?: string; richText?: TLRichText }): void
+	handleInputPointerDown(e: React.PointerEvent<HTMLElement>): void
+	handleDoubleClick(e: any): any
+}
+
+/**
+ * N.B. In Development mode you need to ensure you're testing this without StrictMode on.
+ * Otherwise it's not gonna work as expected on iOS.
+ */
```

Comment:
when we have strict mode issues, we semi-regularly get bug reports as a result. What does this issue look like? Do we have any fix ideas?

Reply:
I'll update the comment - it means that the virtual keyboard won't pop open sometimes (iOS starts flipping out when you render multiple times when trying to focus something)